server-spi/src/main/java/org/keycloak/storage/group/GroupLookupProvider.java

```diff
@@ -83,13 +82,26 @@ default Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String s
      * This is done until the group node does not have a parent (root group)
      *
      * @param realm Realm.
-     * @param search Case sensitive searched string.
-     * @param exact Boolean which defines wheather search param should be matched exactly.
+     * @param search searched string.
+     * @param exact Boolean which defines wheather search param should be matched exactly (including case).
      * @param firstResult First result to return. Ignored if negative or {@code null}.
      * @param maxResults Maximum number of results to return. Ignored if negative or {@code null}.
      * @return Stream of root groups that have the given string in their name themself or a group in their child-collection has.
      * The returned hierarchy contains siblings that do not necessarily have a matching name. Never returns {@code null}.
      */
     Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String search, Boolean exact, Integer firstResult, Integer maxResults);
 
+    /**
+     * Returns the subgroups of a given group id
+     * @param realm Realm
+     * @param id the id of parent group to be used to load subgroups with
+     * @param firstResult First result to return, Ignored if negative or {@code null}.
+     * @param maxResults Maximum number of results to return. Ignored if negative or {@code null}.
+     * @return A Stream of groups that all have the given parent group as their parent.
+     */
+    default Stream<GroupModel> searchForSubgroupsByParentIdStream(RealmModel realm, String id, Integer firstResult, Integer maxResults) {
+        return searchForSubgroupsByParentIdNameStream(realm, id, "", false, firstResult, maxResults);
+    }
+
+    Stream<GroupModel> searchForSubgroupsByParentIdNameStream(RealmModel realm, String id, String search, Boolean exact, Integer firstResult, Integer maxResults);
```

Comment:
This method is a breaking change for public spis. We should document this or provide a default implementation.

Reply:
I have added documentation about this new method in lieu of a default implementation. As far as I can tell, the only default possible is returning an empty stream which I think would cause confusion for implementors more than the breaking change